app/kuma-dp/pkg/dataplane/accesslogs/streamer.go

```diff
@@ -0,0 +1,147 @@
+//go:build !windows
+
+package accesslogs
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"os"
+	"sync"
+	"syscall"
+
+	"github.com/pkg/errors"
+
+	kumadp "github.com/kumahq/kuma/pkg/config/app/kuma-dp"
+	"github.com/kumahq/kuma/pkg/core"
+	"github.com/kumahq/kuma/pkg/core/runtime/component"
+	"github.com/kumahq/kuma/pkg/xds/envoy"
+)
+
+var logger = core.Log.WithName("access-log-streamer")
+
+var _ component.Component = &accessLogStreamer{}
+
+// accessLogStreamer implements TCP access logging in Kuma.
+// When TCP logging is configured, CP configures Envoy to log access logs to a pipe file.
+// Format of such logs is "address:port;msg"
+// Streamer then reads logs from the pipe and passes it to a TCP destination.
+type accessLogStreamer struct {
+	address string
+
+	sync.RWMutex
+	senders map[string]*logSender
+}
+
+func (s *accessLogStreamer) NeedLeaderElection() bool {
+	return false
+}
+
+func NewAccessLogStreamer(dataplane kumadp.Dataplane) *accessLogStreamer {
+	return &accessLogStreamer{
+		address: envoy.AccessLogSocketName(dataplane.Name, dataplane.Mesh),
+		senders: map[string]*logSender{},
+	}
+}
+
+func (s *accessLogStreamer) Start(stop <-chan struct{}) error {
+	logger.Info("cleaning existing access log pipe", "file", s.address)
+	err := os.Remove(s.address)
+	if err != nil && !os.IsNotExist(err) {
+		return errors.Wrapf(err, "error removing existing fifo %s", s.address)
+	}
+	logger.Info("creating access log pipe", "file", s.address)
+	err = syscall.Mkfifo(s.address, 0666)
+	if err != nil {
+		return errors.Wrapf(err, "error creating fifo %s", s.address)
+	}
+	fd, err := os.OpenFile(s.address, os.O_CREATE, os.ModeNamedPipe)
+	if err != nil {
+		return errors.Wrapf(err, "error opening fifo %s", s.address)
+	}
+
+	reader := bufio.NewReader(fd)
+
+	defer func() {
+		fd.Close()
+		s.cleanup()
+	}()
+
+	logger.Info("starting log streamer", "file", s.address)
+	errCh := make(chan error, 1)
+	go func() {
+		if err := s.streamAccessLogs(reader); err != nil {
+			errCh <- err
+		}
+	}()
+	select {
+	case err := <-errCh:
+		return errors.Wrap(err, "log streamer terminated with an error")
+	case <-stop:
+		logger.Info("stopping log streamer")
+		return nil
+	}
+}
+
+func (s *accessLogStreamer) cleanup() {
+	s.RLock()
+	defer s.RUnlock()
+	for _, sender := range s.senders {
+		logger.Info("closing connection to the TCP log destination", "address", sender)
+		if err := sender.close(); err != nil {
```

Comment:
maybe also delete it from the `senders` map after closing, otherwise, after ResilientComponent restarts, `senders` won't be empty

Reply:
Good point! Fixed. I clear the whole map now.